plugins/woocommerce/src/Internal/Admin/Logging/PageController.php

```diff
@@ -305,28 +463,67 @@ private function handle_list_table_bulk_actions(): void {
 			wp_safe_redirect( remove_query_arg( $removable_args, $request_uri ) );
 			exit;
 		}
+	}
 
-		$deleted = filter_input( INPUT_GET, 'deleted', FILTER_VALIDATE_INT );
+	/**
+	 * Format a log file line.
+	 *
+	 * @param string $text        The unformatted log file line.
+	 * @param int    $line_number The line number.
+	 *
+	 * @return string
+	 */
+	private function format_line( string $text, int $line_number ): string {
+		$classes = array( 'line' );
 
-		if ( is_numeric( $deleted ) ) {
-			add_action(
-				'admin_notices',
-				function() use ( $deleted ) {
-					?>
-					<div class="notice notice-info is-dismissible">
-						<p>
-							<?php
-							printf(
-								// translators: %s is a number of files.
-								esc_html( _n( '%s log file deleted.', '%s log files deleted.', $deleted, 'woocommerce' ) ),
-								esc_html( number_format_i18n( $deleted ) )
-							);
-							?>
-						</p>
-					</div>
-					<?php
-				}
+		$level_severities = range( 100, 800, 100 );
+		$severity_levels  = array();
+		foreach ( $level_severities as $severity ) {
+			$severity_levels[] = WC_Log_Levels::get_severity_level( $severity );
+		}
+
+		$text = esc_html( trim( $text ) );
+		if ( empty( $text ) ) {
+			$text = '&nbsp;';
+		}
+
+		$segments = explode( ' ', $text, 3 );
+
+		if ( isset( $segments[0] ) && false !== strtotime( $segments[0] ) ) {
+			$classes[]   = 'log-entry';
+			$segments[0] = sprintf(
+				'<span class="log-timestamp">%s</span>',
+				$segments[0]
 			);
 		}
+
+		if ( isset( $segments[1] ) && in_array( strtolower( $segments[1] ), $severity_levels, true ) ) {
+			$segments[1] = sprintf(
+				'<span class="%1$s">%2$s</span>',
+				esc_attr( 'log-level log-level--' . strtolower( $segments[1] ) ),
+				esc_html( $segments[1] )
+			);
+		}
+
+		if ( count( $segments ) > 1 ) {
+			$text = implode( ' ', $segments );
+		}
+
+		$classes = implode( ' ', $classes );
+		$line    = sprintf(
+			'<span id="L%1$d" class="%2$s">%3$s%4$s</span>',
+			absint( $line_number ),
+			esc_attr( $classes ),
+			sprintf(
+				'<a href="#L%1$d" class="line-anchor"></a>',
+				absint( $line_number )
+			),
+			sprintf(
+				'<span class="line-content">%s</span>',
+				wp_kses_post( $text )
+			)
+		);
+
+		return $line;
```

Comment:
Also non-essential, just something to consider (remove an unnecessary assignment):

```suggestion

		return sprintf(
			'<span id="L%1$d" class="%2$s">%3$s%4$s</span>',
			absint( $line_number ),
			esc_attr( $classes ),
			sprintf(
				'<a href="#L%1$d" class="line-anchor"></a>',
				absint( $line_number )
			),
			sprintf(
				'<span class="line-content">%s</span>',
				wp_kses_post( $text )
			)
		);
```